src/components/input/_index.scss

```diff
@@ -0,0 +1,81 @@
+@use "../../scss";
+
+.c-input {
+	@include scss.component-properties("input");
+
+	.c-input__input {
+		@include scss.component-properties("input-input");
+	}
+
+	.c-input__label {
+		@include scss.component-properties("input-label");
+	}
+
+	.c-input__tip {
+		@include scss.component-properties("input-tip");
+	}
+}
+
+.c-input--error {
+	@include scss.component-properties("input-error");
+
+	.c-input__input {
+		@include scss.component-properties("input-error-input");
+	}
+
+	.c-input__label {
+		@include scss.component-properties("input-error-label");
+	}
+
+	.c-input__tip {
+		@include scss.component-properties("input-error-tip");
+	}
+}
+
+.c-input--success {
+	@include scss.component-properties("input-success");
+
+	.c-input__input {
+		@include scss.component-properties("input-success-input");
+	}
+
+	.c-input__label {
+		@include scss.component-properties("input-success-label");
+	}
+
+	.c-input__tip {
+		@include scss.component-properties("input-success-tip");
+	}
+}
+
+.c-input--warning {
+	@include scss.component-properties("input-warning");
+
+	.c-input__input {
+		@include scss.component-properties("input-warning-input");
+	}
+
+	.c-input__label {
+		@include scss.component-properties("input-warning-label");
+	}
+
+	.c-input__tip {
+		@include scss.component-properties("input-warning-tip");
+	}
+}
+
+.c-input:focus {
```

Comment:
`:focus` is a state only applicable to `<input />` in this component we should update the Scss to have it applied to the input - similar to this

```scss
	.c-input__input {
		@include scss.component-with-states("input-warning-input", [focus]);
	}
```